### DISCUSSION TITLE
Can `RedirectResponse` return flash messages or context like `TemplateResponse`?

### DISCUSSION THREAD

**Discussion by danbailo** [fastapi/fastapi#6088] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.get('')
async def logout():
    response = RedirectResponse(
        '/login', 302
    )
    response.delete_cookie(key='access_token')
    return response
```

### Description

I would to return flash messages after redirect an request. Actually, I do it using `TemplateResponse` and would be great if we can do it in `RedirectResponse` too!

### Wanted Solution

A simple solution would be if we can return a `msg` arg inside `RedirectResponse` or return the context(`Request`) and pass it inside.

### Wanted Code

```python
@router.get('')
async def logout():
    response = RedirectResponse(
        '/login', 302, msg='Sucessfully logout!'
    )
    response.delete_cookie(key='access_token')
    return response
```

### Operating System

Linux

### FastAPI Version

0.88.0

### Python Version

Python 3.11.0

**Comment by jgould22** [comment]
I doubt this is common enough when using a redirect that it would warrant inclusion into the api (but maybe I am wrong)

The good news however is you can make this pretty trivially yourself by sub classing [RedirectResponse](https://github.com/encode/starlette/blob/d7558515020f4c0a96b199880c181a40678b18d5/starlette/responses.py#L204) and then using [set_cookie](https://github.com/encode/starlette/blob/d7558515020f4c0a96b199880c181a40678b18d5/starlette/responses.py#L103) to set the cookies you need for your flash (if that is how you are doing it)

**Comment by danbailo** [comment, marked as answer]
> I doubt this is common enough when using a redirect that it would warrant inclusion into the api (but maybe I am wrong)
> 
> The good news however is you can make this pretty trivially yourself by sub classing [RedirectResponse](https://github.com/encode/starlette/blob/d7558515020f4c0a96b199880c181a40678b18d5/starlette/responses.py#L204) and then using [set_cookie](https://github.com/encode/starlette/blob/d7558515020f4c0a96b199880c181a40678b18d5/starlette/responses.py#L103) to set the cookies you need for your flash (if that is how you are doing it)

I solved it returning the flash messagens in headers and implemented a middleware to set the values in scope of the requests to render the values in jinja templates.
